lib/Backend/Lower.cpp

```diff
@@ -3784,11 +3800,15 @@ Lowerer::GenerateProfiledNewScArrayFastPath(IR::Instr *instr, Js::ArrayCallSiteI
         uint const offsetStart = sizeof(Js::SparseArraySegmentBase);
         headOpnd = GenerateArrayLiteralsAlloc<Js::JavascriptArray>(instr, &size, arrayInfo, &isZeroed);
         const IR::AutoReuseOpnd autoReuseHeadOpnd(headOpnd, func);
+
+        IR::Opnd * missingItemOpnd = getMissingItemOpnd(TyVar);
+        const IR::AutoReuseOpnd autoReuseMissingItemOpnd(missingItemOpnd, func);
+
         for (; i < size; i++)
         {
             GenerateMemInit(
                 headOpnd, offsetStart + i * sizeof(Js::Var),
-                IR::AddrOpnd::New(Js::JavascriptArray::MissingItem, IR::AddrOpndKindConstantAddress, m_func, true),
+                missingItemOpnd,
                 instr, isZeroed);
         }
```

Comment:
nit: These three blocks look extremely similar; is there potentially a benefit to patterning them out to three calls to a single lambda?

Reply:
Yeah, that's a good idea. done.